### PR TITLE
v0.6.0 Update LineItem Array to use Map

### LINKED ISSUE #17: Line item array data structure update
Refactor the line item data structure to improve lookups that is still flexible with add/update/delete

One idea: convert it to a map with the UID as key and line item details as values

### PR DESCRIPTION
**Pull request by michaelharlow**
- Changed LineItem array to use Map structure
- removing LineItems now uses the key assigned rather than searching for a match
Resolves: https://github.com/trandrew1023/edi-translator/issues/17

image

### CODE REVIEW

**Comment by trandrew1023** [comment]
Can you update the following version to 0.6.0 in the package.json https://github.com/trandrew1023/edi-translator/blob/e5dae8fdc97f5431e77ff45218abad850c347119/package.json#L3

File: src/App.jsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ function App() {
           )}
         </Grid>
       </Box>
-    </ThemeProvider>
+    </ThemeProvider>
```
Comment (trandrew1023) [inline src/App.jsx]: nit: this space can be removed to avoid changes to this file

File: src/common/855Translator.js [hunk 2]
```diff
@@ -48,7 +48,8 @@ export function to855(purchaseOrder, lineItems) {
     text = text.concat(`\\`);
     numberOfSegments++;
   }
-  lineItems.forEach((lineItem, index) => {
+  let index = 0;
+  lineItems.forEach((lineItem) => {
```
Comment (trandrew1023) [inline src/common/855Translator.js:52]: what's the reason for creating the index variable?

Comment (michaelharlow) [reply src/common/855Translator.js:52]: The forEach() method on a map doesn't pass the index into the function like the forEach() method on an array so I just pulled it out above

Comment (trandrew1023) [reply src/common/855Translator.js:52]: Gotcha thanks for explaining! lets start the index at 1 then on line 51, then on line 54 we can do index++ and remove the index++ on line 83

**Review by trandrew1023 — APPROVED** [review]

**Pull request merged**